libs/infinity_emb/infinity_emb/transformer/vision/utils.py

```diff
@@ -49,13 +51,50 @@ async def resolve_from_img_url(
             f"error opening the payload from an image in your request from url: {e}"
         )
 
+def resolve_from_img_base64(uri: str) -> ImageSingle:
+    """Resolve an image from a Data URI"""
+    try:
+        base64_image = uri.split(",")[-1]
+        decoded_image = b64decode(base64_image)
+        img = Image.open(io.BytesIO(decoded_image))
+        return ImageSingle(image=img)
+    except Exception as e:
+        raise ImageCorruption(
+            f"error decoding data URI: {e}"
+        )
+
+
+def is_base64_check(s: str):
+    """Regex check to quickly check if string is base64 or not."""
+    pattern = (
+        r"^[A-Za-z0-9+/]{4}([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$"
```

Comment:
Is this a performance efficient check for this? Is it needed (doesn't the Mozilla spec say it needs to say base64?